clients/src/test/java/org/apache/kafka/clients/producer/RecordSendTest.java

```diff
@@ -75,24 +77,17 @@ public void testError() throws Exception {
      */
     @Test
     public void testBlocking() throws Exception {
-        FutureRecordMetadata future = new FutureRecordMetadata(asyncRequest(baseOffset, null, 50L),
-                relOffset, RecordBatch.NO_TIMESTAMP, 0L, 0, 0);
+        time = Time.SYSTEM;
+        FutureRecordMetadata future = new FutureRecordMetadata(produceRequestResult(baseOffset, null),
+                relOffset, RecordBatch.NO_TIMESTAMP, 0L, 0, 0, time);
         assertEquals(baseOffset + relOffset, future.get().offset());
     }
 
     /* create a new request result that will be completed after the given timeout */
-    public ProduceRequestResult asyncRequest(final long baseOffset, final RuntimeException error, final long timeout) {
+    private ProduceRequestResult produceRequestResult(final long baseOffset, final RuntimeException error) {
```

Comment:
This comment needs to be updated since we are not passing a `timeout`. What's the intent behind this change?

Reply:
Here the method started a separate thread and played with Thread.sleep which makes tests slower and flakier.
I had to modify the test class when introduced Time interface in FutureRecordMetadata class, and it made me to change this tests too to be faster/more efficient.

Reply:
It seems like you're changing the semantics of the tests though. It may be best to leave them as they were.

Reply:
It seems like you're changing the semantics of the tests though. It may be best to leave them as they were.

Reply:
ok, i c. 
reverted